EventsExpress.Core/Services/EventService.cs

```diff
@@ -58,12 +58,58 @@ public async Task<OperationResult> AddUserToEvent(Guid userId, Guid eventId)
                 ev.Visitors = new List<UserEvent>();
             }
 
-            ev.Visitors.Add(new UserEvent { EventId = eventId, UserId = userId });
+            if (ev.IsPublic)
+            {
+                ev.Visitors.Add(new UserEvent { EventId = eventId, UserId = userId, UserStatusEvent = 0 });
+            }
+            else
+            {
+                ev.Visitors.Add(new UserEvent { EventId = eventId, UserId = userId, UserStatusEvent = (UserStatusEvent)2 });
+            }
+
             await _db.SaveAsync();
 
             return new OperationResult(true);
         }
 
+        public async Task<OperationResult> ApproveUserToEvent(Guid userId, Guid eventId, bool action)
```

Comment:
Let's name it a `ChangeVisitorStatus` and pass in instead pf `bool action` a `UserStatusEvent status`.